ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/observer/DelegatedCall.kt

```diff
@@ -1,106 +1,59 @@
-*/
+ * Copyright 2014-2025 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
 
 package io.ktor.client.plugins.observer
 
-import io.ktor.client.*
 import io.ktor.client.call.*
-import io.ktor.client.request.*
-import io.ktor.client.statement.*
 import io.ktor.http.*
-import io.ktor.util.date.*
 import io.ktor.utils.io.*
-import kotlin.coroutines.CoroutineContext
 
 /**
  * Wrap existing [HttpClientCall] with new [content].
  *
+ * **Warning:** The content of the returned call response is non-replayable, so it can be consumed only once.
+ * Consider using `wrapWithContent` with lambda instead.
+ *
  * [Report a problem](https://ktor.io/feedback/?fqname=io.ktor.client.plugins.observer.wrapWithContent)
  */
-public fun HttpClientCall.wrapWithContent(content: ByteReadChannel): HttpClientCall {
-    return DelegatedCall(client, content, this)
-}
+@Deprecated(
+    "Use 'withResponse' instead.",
+    ReplaceWith(
+        "withResponse { content }",
+        "io.ktor.client.call.withResponse"
+    )
+)
+public fun HttpClientCall.wrapWithContent(content: ByteReadChannel): HttpClientCall = withResponse { content }
 
 /**
  * Wrap existing [HttpClientCall] with new content produced by the given [block].
+ * The [block] will be called each time the response content is requested.
  *
  * [Report a problem](https://ktor.io/feedback/?fqname=io.ktor.client.plugins.observer.wrapWithContent)
  */
-public fun HttpClientCall.wrapWithContent(block: () -> ByteReadChannel): HttpClientCall {
-    return DelegatedCall(client, block, this)
-}
+@Deprecated(
+    "Use 'withResponse' instead.",
+    ReplaceWith(
+        "withResponse { block() }",
+        "io.ktor.client.call.withResponse"
+    )
+)
+public fun HttpClientCall.wrapWithContent(block: () -> ByteReadChannel): HttpClientCall = withResponse { block() }
 
 /**
  * Wrap existing [HttpClientCall] with new response [content] and [headers].
  *
- * [Report a problem](https://ktor.io/feedback/?fqname=io.ktor.client.plugins.observer.wrap)
- */
-public fun HttpClientCall.wrap(content: ByteReadChannel, headers: Headers): HttpClientCall {
-    return DelegatedCall(client, content, this, headers)
-}
-
-/**
- * Wrap existing [HttpClientCall] with new [headers] and content produced by the given [block].
- * The [block] will be called each time the response content is requested.
- *
- * ```
- * // Example: Content decompression. See ContentEncoding implementation for full example
- * val originalContent = originalCall.response.content
- * val decodedCall = originalCall.wrap(headersOf("Content-Encoding", "identity")) {
- *     decodeGzip(originalContent) // returns a new ByteReadChannel with decoded content
- * }
- * ```
+ * **Warning:** The content of the returned call response is non-replayable, so it can be consumed only once.
+ * Consider using `wrapWithContent` with lambda instead.
```

Comment:
Oh no. I've noticed another typo here. It should be:
```suggestion
 * **Warning:** The content of the returned call response is non-replayable, so it can be consumed only once.
 * Consider using [withResponse].
```